fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -1644,14 +1670,17 @@ private PlanFragment visitPhysicalJoin(PlanFragment leftFragment, PlanFragment r
                             rightFragmentPlanRoot.canDoReplicatedJoin()) {
                         distributionMode = JoinNode.DistributionMode.REPLICATED;
                     } else if (isShuffleJoin(optExpr)) {
```

Comment:
the right child of bucket shuffle HJ is ExchangNode, so isShuffleJoin(optExpr) should always return false, it is dead code.